pkg/plugin/plugin.go

```diff
@@ -122,7 +122,7 @@ func setupVeth(contNetns ns.NetNS, contIfaceName string, requestedMac string, mt
 	// this we will make sure that both ends of the veth pair will be removed
 	// when the container is gone.
 	err := contNetns.Do(func(hostNetns ns.NetNS) error {
-		hostVeth, containerVeth, err := ip.SetupVeth(contIfaceName, mtu, hostNetns)
+		hostVeth, containerVeth, err := ip.SetupVeth(contIfaceName, mtu, requestedMac, hostNetns)
```

Comment:
This already returns the current state of the container, right? I think we can replace the `containerLink` lookup with it.

Reply:
Ah, nice ! Fixed it.